### PR TITLE
[cpp-httplib] update to 0.15.0

### LINKED ISSUE #36423: [cpp-httplib] update to v0.15.0
### Library name

cpp-httplib

### New version number

v0.15.0

### Other information that may be useful (release notes, etc...)

[v0.15.0 release](https://github.com/yhirose/cpp-httplib/releases/tag/v0.15.0)

### PR DESCRIPTION
**Pull request by miyanyan**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/cpp-httplib/fix-find-brotli.patch [hunk 1]
```diff
@@ -1,39 +1,40 @@
...
+@@ -275,9 +275,6 @@ if(HTTPLIB_INSTALL)
+ 	install(FILES
+ 		"${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+ 		"${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+-		# Install it so it can be used later by the httplibConfig.cmake file.
+-		# Put it in the same dir as our config file instead of a global path so we don't potentially stomp on other packages.
+-		"${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindBrotli.cmake"
+ 		DESTINATION ${_TARGET_INSTALL_CMAKEDIR}
+ 	)
+
```
Comment (LilyWangLL) [inline ports/cpp-httplib/fix-find-brotli.patch:40]: Please change the EOL from CRLF to LF.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
All features passed with following triplets:
```
x86-windows
x64-windows
x64-windows-static
```

**Comment by MonicaLiu0311** [comment]
Fixes https://github.com/microsoft/vcpkg/pull/36438.

**Comment by miyanyan** [comment]
there's a new release https://github.com/yhirose/cpp-httplib/releases/tag/v0.15.1

**Review by LilyWangLL — DISMISSED** [review]

**Pull request merged**